### PR TITLE
Pin dependencies to 3.1.0 (for now)

### PR DESCRIPTION
**Pull request by bricelam**
cc @aspnet/build

### CODE REVIEW

File: NuGet.config [hunk 1]
```diff
@@ -2,6 +2,7 @@
 <configuration>
   <packageSources>
     <clear />
+    <add key="aspnet-extensions" value="https://dotnetfeed.blob.core.windows.net/aspnet-extensions/index.json" />
```
Comment (smitpatel) [inline NuGet.config]: Do we need all these feeds if dependencies are pinned?

Comment (bricelam) [reply NuGet.config]: I kept everything mostly intact until we decide on a strategy. E.g. We could continue flowing Extensions but stay on the latest public preview of .NET 5

Comment (bricelam) [reply NuGet.config]: We need this particular feed because one of the packages we depend on isn't published to NuGet.org

**Comment by wtgodbe** [comment]
@bricelam why was this necessary? This means that AspNetCore will never update anything from Extensions (or below), since it transitively gets those dependencies through EFCore: https://github.com/aspnet/AspNetCore/blob/bd7c89aa201c4bf80da2f02f920f1bbbd7109e06/eng/Version.Details.xml#L60-L294

**Comment by Pilchie** [comment]
We're deliberately pulling EntityFrameworkCore out of dependency flow to treat it more like an external package.  We'll have to update the subscription to put Extensions directly into AspNetCore.

**Comment by dougbu** [comment]
@Pilchie nit: Not directly from Extensions but via AspNetCore-Tooling instead of EntityFrameworkCore

**Comment by Pilchie** [comment]
@dougbu for now - until we merge AspNetCore-Tooling into AspNetCore.

**Comment by dougbu** [comment]
Absolutely 😺

**Pull request merged**